src/GitVersion.Core.Tests/IntegrationTests/DevelopScenarios.cs

```diff
@@ -54,13 +54,13 @@ public void WhenDevelopBranchedFromTaggedCommitOnMainVersionDoesNotChange()
     [Test]
     public void CanChangeDevelopTagViaConfig()
     {
-        var config = new Config
+        var configuration = new GitVersionConfiguration
```

Comment:
I agree we need to change the behavior of how to passing in the configuration (change partial to complete). At the moment the TestGitVersionConfigurationBuilder creates per default the git flow configuration you can override. When you define additional branches it gives an error. In my opinion we need to change it and support other branches as well. But before invest time I would like to point out some problems in a productive scenario:

If you want to override e.g. the BranchName of a feature branch with the value null then this is not possible because in the code we are not distinguish a property which hasn't been set and a property which is null.

Reply:
But before I'm going to invest time I would like to point out some problems in a productive scenario:

If you want to override e.g. the Tag of a feature branch (defaults to {BranchName}) with the value null then this is not possible because in the code we are not distinguish a property which hasn't been set and a property which is null.

```csharp
var initalConfiguration = TestConfigurationBuilder.New.WithoutTag("feature").Build();
initalConfiguration.Branches["feature"].Tag.ShouldBe(null); // okay
var configurationBuilder = new ConfigurationBuilder();
configurationBuilder.Add(initalConfiguration);
var configuration = configurationBuilder.Build();
configuration.Branches["feature"].Tag.ShouldBe(null); // exception because Tag
```

Reply:
I would say not to do anything in this PR and pick it up in this issue https://github.com/GitTools/GitVersion/issues/2388. What do you think?

Reply:
Agreed.